[ao] Update DynamicStatic Detector to account for Conv

vspenubarthi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81975
* #81974
* #81973
* #81673
* __->__ #81972
* #81971

Summary: This updates the DynamicStatic Detector to also provide insight
into whether Conv layers should use dynamic or static quantization.
Before, this was not included because as of now, Dynamic quantization is
not supported for Conv layers, but this adds a check for Conv layers and
if dynamic is recommended, it will also give a disclaimer that it is not
currently supported but will be in the future.

Test Plan: python test/test_quantization.py TestFxModelReportDetectDynamicStatic

Reviewers:

Subscribers:

Tasks:

Tags:

vspenubarthi committed:
[ao] Update DynamicStatic Detector to account for Conv

Summary: This updates the DynamicStatic Detector to also provide insight
into whether Conv layers should use dynamic or static quantization.
Before, this was not included because as of now, Dynamic quantization is
not supported for Conv layers, but this adds a check for Conv layers and
if dynamic is recommended, it will also give a disclaimer that it is not
currently supported but will be in the future.

Test Plan: python test/test_quantization.py TestFxModelReportDetectDynamicStatic

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi committed:
Update on "[ao] Update DynamicStatic Detector to account for Conv"


Summary: This updates the DynamicStatic Detector to also provide insight
into whether Conv layers should use dynamic or static quantization.
Before, this was not included because as of now, Dynamic quantization is
not supported for Conv layers, but this adds a check for Conv layers and
if dynamic is recommended, it will also give a disclaimer that it is not
currently supported but will be in the future.

Test Plan: python test/test_quantization.py TestFxModelReportDetectDynamicStatic

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/fx/_model_report/detector.py:
```diff
@@ -460,6 +476,9 @@ def generate_detector_report(self, model: GraphModule) -> Tuple[str, Dict[str, A
 
+        dynamic_benefit = " You will get more accurate results if you use dynamic quantization"
+        static_benefit = " You can increase model efficiency if you use static quantization"
+        future_support_str = ". This layer is not yet supported for dynamic quantization but will be soon"
```
nit: maybe remove `will be soon`, we are not sure when we'll support dynamic conv actually

jerryzh168 commented on torch/ao/quantization/fx/_model_report/detector.py:
```diff
@@ -522,7 +542,7 @@ def generate_detector_report(self, model: GraphModule) -> Tuple[str, Dict[str, A
 
         if not modules_added:
-            dynamic_vs_static_string += "No applicable layers for suggestions. Only linear valid.\n"
+            dynamic_vs_static_string += "No applicable layers for suggestions. Only linear and conv valid.\n"
```
nit: `are valid`

vspenubarthi committed:
Update on "[ao] Update DynamicStatic Detector to account for Conv"


Summary: This updates the DynamicStatic Detector to also provide insight
into whether Conv layers should use dynamic or static quantization.
Before, this was not included because as of now, Dynamic quantization is
not supported for Conv layers, but this adds a check for Conv layers and
if dynamic is recommended, it will also give a disclaimer that it is not
currently supported but will be in the future.

Test Plan: python test/test_quantization.py TestFxModelReportDetectDynamicStatic

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vspenubarthi commented:
@pytorchbot merge -g
